final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,174 @@
+        super().__init__(msg)
+
+
+def is_connected() -> bool:
+    log.info("Checking internet connection...")
+    try:
+        socket.create_connection(("www.google.com", 80))
+        return True
+    except OSError:
+        pass
+    raise CustomEx("No internet connection")
+
+
+def find_news(url) -> bs4.element.ResultSet:
+    """Parses news"""
+    logging.info("Searching for news on " + url)
+    try:
+        resp = requests.get(url)
+        soup = BeautifulSoup(resp.content, features="xml")
+        items = soup.findAll('item')
+        log.info(str(len(items)) + " pieces of news have been found")
+    except requests.exceptions.MissingSchema:
+        log.error("Wrong URL")
+        raise CustomEx("Wrong URL: " + url)
+    except requests.exceptions.ConnectionError:
+        log.error("Can't connect to the " + url)
+        raise CustomEx("Can't connect to the " + url)
+    return items
+
+
+def find_channel(url) -> bs4.element.Tag:
+    logging.info("Getting feed from " + url)
+    try:
+        resp = requests.get(url)
+        soup = BeautifulSoup(resp.content, features="xml")
+        channel = soup.find('channel')
+    except requests.exceptions.ConnectionError:
+        log.error("Wrong URL")
+        raise CustomEx("Wrong URL: " + url)
+    return channel
+
+
+def collect_news(items, limit) -> list:
+    """Sorts the components of the news into dictionaries and adds them into a list"""
+    log.info("Collecting news...")
+    news = []
+    for item in items:
+        try:
+            news_item = {}
+            images = []
+            alts = []
+            news_item['title'] = item.title.text.replace("&#39;", "\'")
+            news_item['date'] = item.pubDate.text
+            soup = BeautifulSoup(item.description.text, 'lxml')
+            news_item['description'] = soup.text.replace("&#39;", "\'")
+            news_item['link'] = item.link.text
+            if item.thumbnail:
+                for img in item.findAll('thumbnail'):
+                    images.append(img['url'])
+            if item.content:
+                for img in item.findAll('content'):
+                    images.append(img['url'])
+            if item.enclosure:
+                for img in item.findAll('enclosure'):
+                    images.append(img['url'])
+            if len(images) == 0:
+                news_item['image'] = None
+            else:
+                news_item['image'] = images
+            if soup.find('img'):
+                for alt in soup.findAll('img'):
+                    alts.append(alt['alt'])
+            if item.find('media:text'):
+                log.info('media:text')
+                soup = BeautifulSoup(item.find('media:text').text, 'lxml')
+                for alt in soup:
+                    alts.append(alt.find('img')['alt'])
+            # if len(item.credit.text) > 0:
+            #     log.info(item.credit.text)
+            #     alts.append(item.credit.text)
+            if len(alts) == 0:
```

Comment:
You can check if the sequence is empty or not simply by using `if `alts`:`, no need to check its length